src/login.rs

```diff
@@ -159,6 +161,24 @@ impl Login {
         self.action_set_enabled("login.next", false);
     }
 
+    pub fn finish_login(&self) {
```

Comment:
I don't really like this as a public API. What about doing that in the `login_client` function?

Reply:
Actually, in the first revision, I had it in the match `AuthorizationState::Ready` match arm. But you are right, it does not need to be exposed.
I had believed that this would not work in this `login_client`. But it looks like it does anyway. It seems my mental model was inaccurate here.